### PR TITLE
dist/IO no indirect, strict & warnings

### PR DESCRIPTION
**Pull request by atoomic**
Make sure dist/IO is not using indirect calls.

### CODE REVIEW

File: dist/IO/lib/IO/Socket.pm [hunk 1]
```diff
@@ -120,7 +120,7 @@ sub connect {
 	if (defined $timeout && ($!{EINPROGRESS} || $!{EWOULDBLOCK})) {
 	    require IO::Select;
 
-	    my $sel = new IO::Select $sock;
+	    my $sel = IO::Select->new( $sock );
```
Comment (toddr) [inline dist/IO/lib/IO/Socket.pm:123]: I've not looked but at one point some of the POD examples also suggested indirect notation

File: dist/IO/t/io_linenum.t [hunk 2]
```diff
@@ -7,7 +7,7 @@ my $File;
 
 BEGIN {
     $File = __FILE__;
-    require strict; import strict;
+    require strict; strict->import();
```
Comment (toddr) [inline dist/IO/t/io_linenum.t:10]: Why not just use strict?

File: dist/IO/lib/IO/Socket.pm (hunk 1 above)
Comment (toddr) [reply dist/IO/lib/IO/Socket.pm:123]: I looked on metacpan. it looks like it's been corrected at this point so we're all good.

**Comment by toddr** [comment]
LGTM

File: dist/IO/t/io_linenum.t (hunk 2 above)
Comment (atoomic) [reply dist/IO/t/io_linenum.t:10]: I try to avoiding refactoring these tests and preserve the original spirit. I agree a `use strict` will work there.

**Review by tonycoz — APPROVED** [review]

**Pull request merged**